### ISSUE TITLE
 False-positive "Use of moved value"

### ISSUE THREAD

**Issue by shekohex** [intellij-rust/intellij-rust#4391]
## Environment

* **IntelliJ Rust plugin version:**  0.2.105.2133-192
* **Rust toolchain version:** 1.39.0-nightly
* **IDE name and version:** Clion 2019.2.1
* **Operating system:** Linux

## Problem description
I got a false positive error "Use of moved value"

## Steps to reproduce
this code compiles without any errors:

```rust
fn reverse(x: i32) -> i32 {
    let mut reversed: i32 = 0;
    let mut n = if x.is_negative() { x.abs() } else { x };
    while n > 0 {
        let last_digit = n % 10;
        reversed = match reversed
            .checked_mul(10)
            .and_then(|val| val.checked_add(last_digit))
        {
            Some(v) => v,
            None => return 0,
        };
        n /= 10;
    }
    if x.is_negative() {
        -reversed
    } else {
        reversed
    }
}
```
Playground: [Click here](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=da78a45614cca98d72b238246a908d5e)

**Comment by vlad20012** [comment]
It seems fixed now.
